HW2/P4/driver.py

```diff
@@ -27,6 +27,22 @@ def py_median_3x3(image, iterations=10, num_threads=1):
 
     return tmpA
 
+def parallelize_py_median_3x3(image, iterations = 10, num_threads = 1):
+    tmpA = image.copy()
+    tmpB = np.empty_like(tmpA)
+    for i in range(iterations):
+        threads = []
+        # Parallel in the median_3x3 call by using the num_threads as step with n as the initial value (offset.)
+        for n in range(num_threads):
+            th = threading.Thread(target = filtering.median_3x3, args = (tmpA, tmpB, n, num_threads, ))
```

Comment:
I found that using Events was very useful, like having red and green lights for threads ! It would have avoided locking everything. But my results are similar to yours !